clients/da-vinci-client/src/main/java/com/linkedin/davinci/schema/merge/PerFieldTimestampMergeRecordHelper.java

```diff
@@ -27,14 +27,11 @@ public UpdateResultStatus putOnField(
     } else if (oldTimestamp == newPutTimestamp) {
       Object oldFieldValue = oldRecord.get(oldRecordField.pos());
       newFieldValue = compareAndReturn(oldFieldValue, newFieldValue, oldRecordField.schema());
-      final boolean newFieldCompletelyReplaceOldField = newFieldValue != oldFieldValue;
-      if (newFieldCompletelyReplaceOldField) {
+
+      if (newFieldValue != oldFieldValue) {
         oldRecord.put(oldRecordField.pos(), newFieldValue);
       }
-      return newFieldCompletelyReplaceOldField
-          ? UpdateResultStatus.COMPLETELY_UPDATED
-          : UpdateResultStatus.NOT_UPDATED_AT_ALL;
-
+      return UpdateResultStatus.COMPLETELY_UPDATED;
```

Comment:
So I guess this is the change referenced in the commit message about:

> **DCR logic must change:**
> Since writes are persisted to rocksdb prior to producing to Kafka, we now must accomodate for the possibility of left over state on a leader. To address this, we add a new mode to the merge conflict resolution logic where upon a perfect tie (on value and timestamp), we resolve to produce the repeated record. The intention here is to be able to be certain that a write which was persisted to rocksdb on leader but not produced doesn't end up getting lost due to failing DCR.

I don't understand this, however... why would there be changes persisted in the leader but not in the VT, in cases of a perfect tie? If it is a perfect tie, then I assume the leader need not persist any changes either, right...?